compiler/cli/bin/kotlinc

```diff
@@ -52,6 +52,10 @@ while [ $# -gt 0 ]; do
       ;;
     -J*)
       java_args=("${java_args[@]}" "${1:2}")
+      if [ "x${1:2}" = "x" ]
+        then echo "error: empty -J argument"
+        exit
```

Comment:
I think we should exit with exit code `1` to signal that the program did not end successfully. Please also check the exit code in the test.

Also, please use the code style equivalent to the rest of the file, where `if` and `then` are on the same line